### ISSUE TITLE
v1.8.1.0 does not actually create virtual devices

### ISSUE THREAD

**Issue by JoshWobbles** [nefarius/ViGEm#39]
After finding out the error 0x20000000 was actually a success and not an error, I am now faced with the issue where ViGEm does not actually "plug in" or create a virtual device when vigem_target_plugin is called. It seems to return VIGEM_ERROR_TARGET_UNINITIALIZED even though ViGemUm.VIGEM_TARGET_INIT(ref _target); is called a couple lines above.

Perhaps VIGEM_TARGET_INIT() could use a return value as well to help track down errors in this portion.

Worth noting I have this issue in both my app and in the ViGEmTester.NET

**Comment by nefarius** [comment]
`VIGEM_TARGET_INIT` is an inline function which is 99.9% impossible to fail 😉 That's why it returns `VOID`.

**Comment by nefarius** [comment]
The native libs work for me with the latest commits.

**Comment by JoshWobbles** [comment]
It was inside the nativelibs class, I assumed it was a pinvoke too quickly. Well something is causing the plugin to return VIGEM_ERROR_TARGET_UNINITIALIZED even though ViGemUm.vigem_init() is returning VigemError.VIGEM_ERROR_NONE

**Comment by nefarius** [comment]
`VIGEM_ERROR_TARGET_UNINITIALIZED` isn't influenced by `vigem_init()`

**Comment by JoshWobbles** [comment]
What could influence it then?

One thing that may be contributing (though I cannot see why as I am using your signed pre built drivers and not building my own) is I am using a newer WDK and had to upgrade the projects... Long shot but only thing I can think of that may be different. Ill try installing the WDK you used aswell and see what happens.

Here is a x64 build of the tester project from here, could you tell me if it works for you, if it does it would rule out the code itself and point me to some sort of environment issue or incompatibility.

Another note, I also still have SCPVB installed, there is no overlap or conflict there is there?

https://1drv.ms/u/s!AnvwSUr7_fTqgts3-JVJ4v8SsJsbJw

**Comment by nefarius** [comment]
Bloody hell, you're right, the .NET version isn't doing jack sh*t 😕 I'll investigate...

**Comment by nefarius** [comment]
No, ViGEm and the ScpVBus can co-exist.

**Comment by JoshWobbles** [comment]
I was really starting to think I was having a stroke or something.

**Comment by nefarius** [comment]
I haven't looked at the managed samples for a few month I have to admit and now that I see the: JESUS CHRIST 🤢

**Comment by JoshWobbles** [comment]
lol

**Comment by nefarius** [comment]
Found it, fixed it, will push it in a few minutes with some other improvements...

**Comment by JoshWobbles** [comment]
Sweet dude ;-) Thanks for taking time for us lowly .net guys lol.

**Comment by nefarius** [comment]
No worries, it was a struct size mismatch due to an update which was hard to spot even if you know the code...

**Comment by nefarius** [comment]
Done & pushed 😈

**Comment by JoshWobbles** [comment]
DS4 works, 360 seems to do nothing, I see the new VID/PID in the struct, do I need to specify 360 controller via that? Right now I am just using var error = ViGemUm.vigem_target_plugin(VigemTargetType.Xbox360Wired, ref _target); with VigemTargetType.Xbox360Wired = 0, and _target being empty other than what is set by the init.

**Comment by JoshWobbles** [comment]
Also for giggles, I tried the Xbox one wired option, and it BSODs

**Comment by nefarius** [comment]
Ah I see you use the native class directly, that's not intended 😉 have a look at `Xbox360Controller` class.

PS: Oooooooops ☘️

**Comment by nefarius** [comment]
Like, this should work, I do the same:

```C#
var x360 = new Xbox360Controller();
x360.PlugIn();
```

**Comment by nefarius** [comment]
Going to sleep now, will implement the missing methods tomorrow and further clean up the classes to avoid confusion.

**Comment by JoshWobbles** [comment]
I'm not seeing the difference, I am using my own class because I am not using the UM wrapper lib due the complexities that involved in other aspects of my build, instead I have my own device.

**Comment by JoshWobbles** [comment]
I uploaded my class, best I can tell the only real difference is I am not using the IViGEmTarget

https://github.com/InputMapper/ViGEM_360/blob/master/ViGEm360Device.cs

**Comment by JoshWobbles** [comment]
Ok, I found the issue, was due to a misunderstanding on how ViGEm works. Apparently it only accepts one application at a time and thus only one initialization. I had 2 libraries, one for DS4 and one for 360 so I could keep thing seperate and apparently both of them calling ViGemUm.vigem_init() was no good. So I merged the libraries and made them inherit a base class with a static constructor so ViGemUm.vigem_init() only gets called once.
